src/storm/storage/jani/TemplateEdge.cpp

```diff
@@ -85,15 +85,16 @@ OrderedAssignments& TemplateEdge::getAssignments() {
     return assignments;
 }
 
-void TemplateEdge::substitute(std::map<storm::expressions::Variable, storm::expressions::Expression> const& substitution) {
-    guard = substituteJaniExpression(guard, substitution);
+void TemplateEdge::substitute(std::map<storm::expressions::Variable, storm::expressions::Expression> const& substitution,
+                              bool const& substituteTranscendentalNumbers) {
+    guard = substituteJaniExpression(guard, substitution, substituteTranscendentalNumbers);
 
     for (auto& assignment : assignments) {
-        assignment.substitute(substitution);
+        assignment.substitute(substitution, substituteTranscendentalNumbers);
```

Comment:
is this maybe something that can be avoided by having a sensible default for this?

Reply:
I am not sure, since this is set in the JaniModel's methods to substitute constants (around [here](https://github.com/moves-rwth/storm/pull/556/files/af3337abf9f01100c009d5e38ee5867ceafd84b9#diff-1271277540852224a7a247ef9c5259a6f3a3bbde0f1d44ba862c8e0f7c9ecb36R1133)) and must be propagated down to the various entries of the model. The value depends whether we want to preserve the constants (e.g. storm-conv) or not (when we run the evaluation).
A way to not have the this argument would be to set a static variable for the jani visitor, to signal whether we want to substitute transcendental numbers or not, but I wouldn't recommend it, since it makes the code execution less clear.